src/vs/workbench/contrib/extensions/node/extensionsWorkbenchService.ts

```diff
@@ -1001,29 +1001,9 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 			if (extension) {
 				return this.windowService.focusWindow()
 					.then(() => this.open(extension));
+			} else {
+				return Promise.resolve(null);
 			}
-
-			return this.queryGallery({ names: [extensionId], source: 'uri' }, CancellationToken.None).then(result => {
```

Comment:
I am afraid you removed too much here. This breaks the functionality if the extension is not installed. If the extension is not installed, you should query from gallery and open the extension. You should only remove the prompt (1015, 1023) after opening the extension.